web/packages/teleport/src/Discover/Server/DiscoveryConfigSsm/DiscoveryConfigSsm.tsx

```diff
@@ -174,94 +196,111 @@ export function DiscoveryConfigSsm() {
-          </ButtonSecondary>
-        )}
-      </StyledBox>
-      {showRestOfSteps && (
+    <Validation>
+      {({ validator }) => (
         <>
+          <Header>Setup Discovery Config for Teleport Discovery Service</Header>
+          {cfg.isCloud ? (
+            <Text>
+              The Teleport Discovery Service can connect to Amazon EC2 and
+              automatically discover and enroll EC2 instances.{' '}
+              <SsmInfoHeaderText />
+            </Text>
+          ) : (
+            <Text>
+              Discovery config defines the setup that enables Teleport to
+              automatically discover and register instances.{' '}
+              <SsmInfoHeaderText />
+            </Text>
+          )}
+          {cfg.isCloud && <SingleEc2InstanceInstallation />}
           <StyledBox mt={4}>
-            <H3>Step 2</H3>
-            <P>
-              Attach AWS managed{' '}
-              <ExternalLink
-                target="_blank"
-                href="https://docs.aws.amazon.com/aws-managed-policy/latest/reference/AmazonSSMManagedInstanceCore.html"
-              >
-                AmazonSSMManagedInstanceCore
-              </ExternalLink>{' '}
-              policy to EC2 instances IAM profile. The policy enables EC2
-              instances to use SSM core functionality.
-            </P>
-          </StyledBox>
-          <StyledBox mt={4}>
-            <H3>Step 3</H3>
-            <P>
-              Each EC2 instance requires{' '}
-              <ExternalLink
-                target="_blank"
-                href="https://docs.aws.amazon.com/systems-manager/latest/userguide/ssm-agent-status-and-restart.html"
-              >
-                SSM Agent
-              </ExternalLink>{' '}
-              to be running. The SSM{' '}
-              <ExternalLink
-                target="_blank"
-                href={`https://${selectedRegion}.console.aws.amazon.com/systems-manager/fleet-manager/managed-nodes?region=${selectedRegion}`}
+            <header>
+              <H3>Step 1</H3>
+              <Subtitle3>
+                Select the AWS Region that contains the EC2 instances that you
+                would like to enroll
+              </Subtitle3>
+            </header>
+            <Box mb={-5}>
+              <AwsRegionSelector
+                onFetch={(region: Regions) => setSelectedRegion(region)}
+                clear={clear}
+                disableSelector={!!scriptUrl}
+              />
+            </Box>
+            {selectedRegion && (
+              <Box mb={3} mt={5}>
+                <P>
+                  You can filter for EC2 instances by their tags. If no tags are
+                  added, Teleport will use a wildcard to enroll all EC2
+                  instances.
+                </P>
+                <LabelsInput
+                  adjective="tag"
+                  labels={tags}
+                  setLabels={setTags}
+                />
+              </Box>
+            )}
```

Comment:
this is the only change to this form. i moved the validator higher which updated the indentation so a lot of diff here that isnt really diff